test/SILOptimizer/diagnose_var_usage.swift

```diff
@@ -0,0 +1,1662 @@
+// RUN: %target-swift-frontend -sil-usage-diagnostics -emit-sil %s -verify
```

Comment:
Could you describe while files this one is based off of? I see there is `decl/var/usage.swift`, then there's other files too. If we can cut down on the size of this file by adding extra `RUN:` lines to other files, it would be valuable, as that would give good signal on what is new vs what already worked.

Reply:
Also, I didn't notice anything here related to concurrency, could be useful to potentially have some tests to make sure the same warnings work correctly when `async let` and `await` are involved.

Reply:
This pulls in tests from a lot of different files, and I'm not sure adding `RUN:` lines will work. Because these diagnostics now occur during SIL optimization, the diagnostics are not produced if the code has errors (because SIL is never generated). So this involved migrating tests from test files that had errors.

I actually have dozens of other test file edits in the original PR, but decided not to merge them yet in order to implement this incrementally.

How do you think I should approach this?